blog/12-12-2022-advance-hardhat-configration-on-celo-using-plugins/advance-hardhat-configration-on-celo-using-plugins.md

```diff
@@ -0,0 +1,751 @@
+- Know how to set up a basic Hardhat project. Please complete the previous example tutorial [here](https://github.com/celo-org/docs/blog/08-12-2022-getting-started-on-celo-with-hardhat) as we’ll also make use of the previous example [here](https://github.com/bobeu/advanced-hardhat-for-celo-dev). Clone the project, and navigate to the project directory.
+
+```bash
+git clone https://github.com/bobeu/advanced-hardhat-for-celo-dev.git
+cd advanced-hardhat-for-celo-dev
+
+```
+
+# Requirements​
+
+To get started, ensure you have the following tools installed on the machine:
+  - Code editor. VSCode recommended.
+  - NodeJs version >=14.0.0. 
+
+*Note:* I am using Node version "18.12.1"
+
+## Hardhat plugins
+
+Continuing from the last tutorial, we will install a few plugins to improve the deployment of the Lock contract to Celo networks and solidify testing.
+An important highlight of Hardhat is the plugins that are external to Hardhat itself. Even though Hardhat comes with some default plugins, you can always override them to use what’s best for your needs. We will discuss some of the external programs - relevant plugins, or the plugins you most likely need. However, this does not include plugins that weren't built before this article was written. During Hardhat installation, If you choose to install the sample project to enable support for waffle tests, “hardhat-waffle” and “hardhat-ethers” are installed alongside “@nomicfoundation/hardhat-toolbox”. As a Celo developer using hardhat, you’ll mostly be running tasks which is a function of the “Hardhat Runner” - a CLI for interacting with Hardhat. Each time you run a Hardhat command from the CLI e.g. “npx hardhat compile”, you’re simply running a task.
+
+Using plugins with Hardhat is one flexibility advantage over Truffle, which simply extends the Hardhat Runtime Environment - HRE . Some of them are in-built, while others are built by the community. Now, let’s examine some of the higher-level libraries rather called *plugins*, and how to integrate and use them.
+
+- @nomicfoundation/hardhat-toolbox
+- @nomicfoundation/hardhat-chai-matchers
+- @openzeppelin/hardhat-upgrades
+- @nomiclabs/hardhat-ethers
+- _@typechain/hardhat_
+- Hardhat-deploy
+- Hardhat-react
+
+Under the hood, these plugins extend Hardhat functionalities by extending the config file. 
+Example:
+
+```bash
+extendEnvironment((hre) => {
+  const Web3 = require("ethers");
+  hre.Web3 = Web3;
+ 
+  hre.web3 = new Web3(hre.network.provider);
+});
+```
+
+Hardhat lets you add instances of these tools to its environment object by installing and initializing them once in "hardhatconfig.js". Soon as they feature in the Hardhat global variables, they’ll be available for use across every Hardhat component. Firstly, we will install and configure the plugins we need, then we will apply the usage. 
+
+*Note:* Hardhat components are files/folders within Hardhat context i.e. within the scope of its reach, such as “test” folder. 
```

Comment:
*Note:* Hardhat components are files/folders within the Hardhat context i.e. within the scope of its reach, such as the “test” folder.